src/components/table-row.tsx

```diff
@@ -337,18 +414,28 @@ const TableRow = ({
     });
   }
 
-  return (<tr
-    className='chaise-table-row'
-    ref={rowContainer}
-    style={{ 'position': 'relative' }}
-  >
-    {showActionButtons &&
-      <td className='block action-btns'>
-        {renderActionButtons()}
-      </td>
-    }
-    {renderCells()}
-  </tr>
+  return (
+  <>
+    <tr
+      className='chaise-table-row'
+      ref={rowContainer}
+      style={{ 'position': 'relative' }}
+    >
+      {showActionButtons &&
+        <td className='block action-btns'>
+          {renderActionButtons()}
+        </td>
+      }
+      {renderCells()}
+    </tr>
+    <DeleteConfirmationModal 
+      show={showDeleteConfirmationModal} 
+      onDeleteConfirmation={onDeleteConfirmation}
+      onCancel={onCancel}
+    >
+      Are you sure you want to delete <DisplayValue value={tuple.displayname} />?
```

Comment:
Instead of passing the children, you should just pass the tuple and let the modal display the body.